runtime/common/src/evm/precompile.rs

```diff
@@ -12,129 +12,66 @@
+impl<Symbol: Get<&'static str>> Erc20Metadata for NativeErc20Metadata<Symbol> {
+	fn name() -> &'static str {
+		Symbol::get()
+	}
 
-impl<R> CentrifugePrecompiles<R> {
-	#[allow(clippy::new_without_default)] // We'll never use Default and can't derive it.
-	pub fn new() -> Self {
-		Self(Default::default())
+	fn symbol() -> &'static str {
+		Symbol::get()
 	}
-}
 
-impl<R> PrecompileSet for CentrifugePrecompiles<R>
-where
-	R: pallet_evm::Config + axelar_gateway_precompile::Config + frame_system::Config,
-	R::RuntimeCall: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo + Decode,
-	<R::RuntimeCall as Dispatchable>::RuntimeOrigin: From<Option<R::AccountId>>,
-	axelar_gateway_precompile::Pallet<R>: Precompile,
-{
-	fn execute(&self, handle: &mut impl PrecompileHandle) -> Option<PrecompileResult> {
-		match handle.code_address().0 {
-			ECRECOVER_ADDR => Some(ECRecover::execute(handle)),
-			SHA256_ADDR => Some(Sha256::execute(handle)),
-			RIPEMD160_ADDR => Some(Ripemd160::execute(handle)),
-			IDENTITY_ADDR => Some(Identity::execute(handle)),
-			MODEXP_ADDR => Some(Modexp::execute(handle)),
-			BN128ADD_ADDR => Some(Bn128Add::execute(handle)),
-			BN128MUL_ADDR => Some(Bn128Mul::execute(handle)),
-			BN128PAIRING_ADDR => Some(Bn128Pairing::execute(handle)),
-			BLAKE2F_ADDR => Some(Blake2F::execute(handle)),
-			SHA3FIPS256_ADDR => Some(Sha3FIPS256::execute(handle)),
-			DISPATCH_ADDR => Some(Dispatch::<R>::execute(handle)),
-			ECRECOVERPUBLICKEY_ADDR => Some(ECRecoverPublicKey::execute(handle)),
-			LP_AXELAR_GATEWAY => {
-				Some(<axelar_gateway_precompile::Pallet<R> as Precompile>::execute(handle))
-			}
-			_ => None,
-		}
+	fn decimals() -> u8 {
+		18
 	}
 
-	fn is_precompile(&self, address: H160, _remaining_gas: u64) -> IsPrecompileResult {
-		IsPrecompileResult::Answer {
-			is_precompile: matches!(
-				address.0,
-				ECRECOVER_ADDR
-					| SHA256_ADDR | RIPEMD160_ADDR
-					| IDENTITY_ADDR | MODEXP_ADDR
-					| BN128ADD_ADDR | BN128MUL_ADDR
-					| BN128PAIRING_ADDR | BLAKE2F_ADDR
-					| SHA3FIPS256_ADDR | DISPATCH_ADDR
-					| ECRECOVERPUBLICKEY_ADDR
-					| LP_AXELAR_GATEWAY
-			),
-			extra_cost: 0,
-		}
+	fn is_native_currency() -> bool {
+		true
 	}
 }
 
-/// Altair's precompiles
-/// For now, Altair uses the exact same set of precompiles used in Development.
-pub type Altair<R> = CentrifugePrecompiles<R>;
-
-/// A set of precompiles. This set might contain
-/// not yet mainnet ready precompiles in order to test
-/// those in development or staging environment without touching
-/// the mainnet set.
-pub type Development<R> = CentrifugePrecompiles<R>;
+type EthereumPrecompilesChecks = (AcceptDelegateCall, CallableByContract, CallableByPrecompile);
 
-// H160 cannot be used in a match statement due to its hand-rolled
-// PartialEq implementation. This just gives a nice name to the
-// internal array of bytes that an H160 wraps.
-type Addr = [u8; 20];
+#[precompile_utils::precompile_name_from_address]
```

Comment:
Using this macro, things can get quite simplified in this module.

Reply:
Thanks for updating to this new format! <3